### PR TITLE
Updates to Scope and Authentication Type on page https://fusionauth.io/docs/v1/tech/oauth/tokens/

### PR DESCRIPTION
**Pull request by jobannon**
# Summary 

Updates to OAuth Token page - add scope in client credentials grant  and AuthenticationType (for newer Idps) 

# Related Issues 

closes https://github.com/FusionAuth/fusionauth-site/issues/784
closes https://github.com/FusionAuth/fusionauth-site/issues/783

### CODE REVIEW

File: site/docs/v1/tech/oauth/tokens.adoc [hunk 1]
```diff
@@ -103,6 +103,10 @@ https://tools.ietf.org/html/rfc7519#section-4.1.1[RFC 7519 Section 4.1.1].
 [field]#permissions# [type]#[Array<String>]#::
 The permission granted to the recipient Entity by the target Entity. This claim is only present if permissions are associated with the grant and any requested permissions are found in the grant.
 
+[field]#scope# [type]#[String]#::
```
Comment (mooreds) [inline site/docs/v1/tech/oauth/tokens.adoc:106]: We handle scopes a bit differently than other providers. Usually a scope is either a standardized string (like `offline_access`, defined in the OIDC spec) or custom, defined by the user of the idp (as outlined here: https://github.com/FusionAuth/fusionauth-issues/issues/275 ). But for this case, we have a specific format we support.

Can you add a link to https://fusionauth.io/docs/v1/tech/oauth/#client-credentials-scopes which outlines what scope values are valid?

File: site/docs/v1/tech/oauth/tokens.adoc [hunk 2]
```diff
@@ -103,6 +103,10 @@ https://tools.ietf.org/html/rfc7519#section-4.1.1[RFC 7519 Section 4.1.1].
...
+The scope of the Access token. See scope defined by
+https://datatracker.ietf.org/doc/html/rfc6749#section-3.3[RFC 6749 Section 3.3] and referenced in Client Credentials Grant https://datatracker.ietf.org/doc/html/rfc6749#section-4.4[RFC 6749 Section 4.4]
+
```
Comment (mooreds) [inline site/docs/v1/tech/oauth/tokens.adoc:109]: You don't need this `+`.

File: site/docs/v1/tech/oauth/tokens.adoc [hunk 3]
```diff
@@ -103,6 +103,10 @@ https://tools.ietf.org/html/rfc7519#section-4.1.1[RFC 7519 Section 4.1.1].
 [field]#permissions# [type]#[Array<String>]#::
 The permission granted to the recipient Entity by the target Entity. This claim is only present if permissions are associated with the grant and any requested permissions are found in the grant.
 
+[field]#scope# [type]#[String]#::
+The scope of the Access token. See scope defined by
```
Comment (mooreds) [inline site/docs/v1/tech/oauth/tokens.adoc:107]:
```suggestion
The scope of the Access token. This meaning of this field is specified by
```

but see above about how we constrain the use of the `scope` parameter.

**Review by mooreds — CHANGES_REQUESTED** [review]
A few changes, pls.

**Comment by mooreds** [comment]
Any progress on this, @jobannon ?

**Comment by jobannon** [comment]
Next after Idp's 🏋🏻 mega merge 🙂

File: site/docs/v1/tech/oauth/tokens.adoc (hunk 2 above)
Comment (jobannon) [reply site/docs/v1/tech/oauth/tokens.adoc:109]: I think this is a goof on Github's part. I show just one space, similar to all other fields under the API class.

**Pull request merged**